### PR TITLE
:sparkles: handle job errors separately from API errors

### PR DESCRIPTION
**Pull request by ianardee**
## Description

Code update following server update.

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires a change to the official Guide documentation.

### CODE REVIEW

File: src/test/java/com/mindee/parsing/common/AsyncPredictResponseTest.java [hunk 1]
```diff
@@ -87,4 +84,18 @@ void whenAsyncGet_returnsCompleted_mustBeDeserialized() throws IOException {
...
+      "src/test/resources/async/get_failed_job_error.json"
+    );
+    Assertions.assertNotNull(response);
+    Assertions.assertEquals("success", response.getApiRequest().getStatus());
+    Assertions.assertEquals(200, response.getApiRequest().getStatusCode());
+    Assertions.assertEquals("failed", response.getJob().getStatus());
+    Assertions.assertEquals("ServerError", response.getJob().getError().getCode());
+    Assertions.assertEquals("2024-02-20T10:31:06.878599", response.getJob().getIssuedAt().toString());
+    Assertions.assertEquals("2024-02-20T10:31:06.878599", response.getJob().getAvailableAt().toString());
+    Assertions.assertFalse(response.getDocument().isPresent());
```
Comment (sebastianMindee) [inline src/test/java/com/mindee/parsing/common/AsyncPredictResponseTest.java:99]: Just curious, isn't testing for .isEmpty() the norm for most cases like these?

**Review by sebastianMindee — COMMENTED** [review]
1 question & 1 suggestion :)

File: src/test/java/com/mindee/parsing/common/AsyncPredictResponseTest.java [hunk 2]
```diff
@@ -40,10 +40,7 @@ void whenAsyncPost_returnsErrorForbidden_mustBeDeserialized() throws IOException
     Assertions.assertNotNull(response);
     Assertions.assertEquals("failure", response.getApiRequest().getStatus());
     Assertions.assertEquals(403, response.getApiRequest().getStatusCode());
-    Assertions.assertNull(response.getJob().getStatus());
-    Assertions.assertEquals("2023-01-01T00:00", response.getJob().getIssuedAt().toString());
-    Assertions.assertNull(response.getJob().getAvailableAt());
-    Assertions.assertFalse(response.getDocument().isPresent());
+    Assertions.assertNull(response.getJob());
```
[the remark this answers is not in the document]
Comment (ianardee) [reply src/test/java/com/mindee/parsing/common/AsyncPredictResponseTest.java:43]: added a check

**Review by sebastianMindee — APPROVED** [review]
Second question about `.isPresent()` still pending, but LGTM code-wise

**Review by sebastianMindee — CHANGES_REQUESTED** [review]
Just noticed, current `code` in the new sample is a String, not an Integer.

File: src/main/java/com/mindee/parsing/common/Job.java [hunk 3]
```diff
@@ -28,27 +28,42 @@
...
+  /**
    * Job Status
    */
   private String status;
 
+  /**
+   * Job Status Code
+   */
+  @JsonProperty("status_code")
+  private Integer statusCode;
```
Comment (sebastianMindee) [inline src/main/java/com/mindee/parsing/common/Job.java]: I don't understand what that is

**Review by sebastianMindee — APPROVED** [review]

**Pull request merged**